token/cli/src/main.rs

```diff
@@ -1891,6 +1911,164 @@ async fn command_sync_native(
     })
 }
 
+// Both enable-required_transfer-mesos and disable-required_transfer-mesos, switches with enable bool
+async fn command_required_transfer_memos(
+    token_account_address: Pubkey,
+    owner: Pubkey,
+    bulk_signers: BulkSigners,
+    enable: bool,
+    config: &Config<'_>,
+) -> CommandResult {
+    let program_id = if config.sign_only {
+        config.program_id
+    } else {
+        config
+            .rpc_client
+            .get_account(&token_account_address)
+            .await
+            .map_err(|err| {
+                format!(
+                    "Token account {} does not exist: {}",
+                    token_account_address, err
+                )
+            })?
+            .owner
+    };
+    let mut instructions: Vec<Instruction> = Vec::new();
+    let reallocate_instruction = reallocate_if_needed(
+        &program_id,
+        &token_account_address,
+        &owner,
+        vec![ExtensionType::MemoTransfer],
+        config,
+    )
+    .await?;
```

Comment:
this should go inside the block that executes when `!config.sign_only` because it depends on rpc. however, both rpc calls inside the `reallocate_if_needed` function are extraneous. we should fetch the token account once and use/clone it as needed

it might be easier to eliminate the function and do the work inline, and we can save refactoring out for the day when its needed by other things (the only other extension i can think of is confidential transfer, and that is going to be a major effort to support)

Reply:
resolved with d37c851